### PR TITLE
Fix for trader crafting assembly build a buff

### PR DESCRIPTION
**Pull request by HeronAlexandria**
The trader assembly buff is applying only food assembly. The intent of this buff is to provide an assembly buff for all kinds of crafting.

Build a buff works in "affects" which indicate what is modified for the selected buff. The existing system allows only for one affect, which is why the assembly buff is limited to food assembly.

This fix adds an mapping table, which allows multiple buffs to be applied to an effect. In this fix, the food_assembly affect was changed to crafting_assembly. The mapping mechanism searches for entries of crafting_assembly, and applies a list of effects. In the case of crafting_assembly, this is the trader assembly buffs.

If the affect is not in the mapping table, then the effect is applied normally. For example, change the effect from crafting_assembly and back to food_assembly and only food assembly will be applied.

Since the fix is table driven, it allows servers to choose how they want the system to function. The effects could be grouped, or a server can choose to apply the assembly buffs individually.

This fix should provide the intended functionality without hard coding.

### CODE REVIEW

File: sku.0/sys.server/compiled/game/script/item/goggles/reset.java [hunk 1]
```diff
@@ -24,6 +24,13 @@ public reset()
     };
     public int createNewGoggles(obj_id self, obj_id target, String params, float defaultTime) throws InterruptedException
     {
+    
+        String config = getConfigSetting("GameServer", "seGogglesReward");
```
Comment (Cekis) [inline sku.0/sys.server/compiled/game/script/item/goggles/reset.java:28]: This has nothing to do with food assembly, lol

File: sku.0/sys.server/compiled/game/script/systems/buff/buff_builder_effect_map.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+        int rowIndex = dataTableSearchColumnForString(affectName, "BUFF_BUILDER_AFFECTS", DATATABLE_BUFF_BUILDER_EFFECT_MAP);
+    
+        if (rowIndex >= 0)
+        {
+            int rowCount = dataTableGetNumRows(DATATABLE_BUFF_BUILDER_EFFECT_MAP);
+
+            for (int index = 0; index < rowCount; index++)
+            {
+                dictionary row = dataTableGetRow(DATATABLE_BUFF_BUILDER_EFFECT_MAP, index);
```
Comment (Cekis) [inline sku.0/sys.server/compiled/game/script/systems/buff/buff_builder_effect_map.java:32]: new instances of objects should be defined OUTSIDE of a loop... then just reuse the object for each iteration.

**Review by Cekis — APPROVED** [review]
Would like QA to test this... @SWG-Source/qa please check out this PR and test to make sure all is working as expected and still inline with original SWG play.

File: sku.0/sys.server/compiled/game/script/item/goggles/reset.java (hunk 1 above)
Comment (HeronAlexandria) [reply sku.0/sys.server/compiled/game/script/item/goggles/reset.java:28]: I added this fix to my repo and it added it to the pull request. Oops.

File: sku.0/sys.server/compiled/game/script/systems/buff/buff_builder_effect_map.java (hunk 2 above)
Comment (HeronAlexandria) [reply sku.0/sys.server/compiled/game/script/systems/buff/buff_builder_effect_map.java:32]: Understood. Will make an update when I can.

**Pull request merged**